File: 1308/1308.c
#include <stdio.h>
#include <math.h>

int main(){
    
    int n,i;
    long long int m;
    double d, r;
    
    scanf("%d", &n);
    
    for(i = 0; i < n; i++){
          scanf("%lld", &m);
          
          //Baskara (a = 1, b= 1, c= -2m)
          r = (-1 + sqrt(1 + 8 * m))/2;
          
          printf("%lld\n", (long long int) r);
    }
    
    return 0;
}

File: 1308/1308.java
import java.io.IOException;
import java.util.Scanner;

public class Main {
    public static void main(String[] args) throws IOException {
        Scanner scanner = new Scanner(System.in);
        
        int n = scanner.nextInt();
        for (int i = 0; i < n; i++) {
            long m = scanner.nextLong();
            
            //Baskara (a = 1, b= 1, c= -2m)
            double r = (-1 + Math.sqrt(1 + 8 * m)) / 2;
            
            System.out.println((int) r);
        }
    }
}